Add external Etcd options to docs (#3937)
k0sproject/k0s: docs/configuration.md

| `etcd.peerAddress` | Node address used for etcd cluster peering.                                                                                                                            |
| `etcd.extraArgs`   | Map of key-values (strings) for any extra arguments to pass down to etcd process.                                                                                      |
| `kine.dataSource`  | [kine](https://github.com/k3s-io/kine) datasource URL.                                                                                                                 |

> | `etcd.externalCluster` | Configuration when Etcd is externally managed, i.e. running on dedicated nodes. See [`spec.storage.etcd.externalCluster`](#specstorageetcdexternalcluster)

Reviewer: The makrdown table lines above also need to be realigned.

```suggestion
| `etcd.externalCluster` | Configuration when etcd is externally managed, i.e. running on dedicated nodes. See [`spec.storage.etcd.externalCluster`](#specstorageetcdexternalcluster)             |
```

#### `spec.storage.etcd.externalCluster`

> k0s can also work with externally managed Etcd cluster. When this is configured, k0s will NOT setup the etcd, user has to manage it.

Reviewer: ```suggestion
K0s can also work with an externally managed etcd cluster. If this is
configured, k0s will NOT set up etcd, it has to be managed manually.
```
